ios/FluentUI/Vnext/List/List.swift

```diff
@@ -53,66 +55,64 @@ public struct MSFListView: View {
+                    ForEach(sections, id: \.self) { section in
+                        if let sectionTitle = section.title {
+                            Header(title: sectionTitle, tokens: tokens)
+                        }
 
-                ForEach(section.cells, id: \.self) { cell in
-                    MSFListCellView(cell: cell,
-                                    tokens: tokens)
-                        .border(section.hasBorder ? Color(tokens.borderColor) : Color.clear, width: section.hasBorder ? tokens.borderSize : 0)
-                        .frame(maxWidth: .infinity)
+                        ForEach(section.cells.indices, id: \.self) { index in
+                            let cell = section.cells[index]
+                            let hasDividers = (index < section.cells.count - 1 && section.hasDividers) || (section.hasDividers && (cell.children != nil))
+                            MSFListCellView(cell: cell,
+                                            tokens: tokens,
+                                            hasDividers: hasDividers)
+                                .frame(maxWidth: .infinity)
+                        }
+                        Divider()
+                            .background(Color(tokens.borderColor))
+                    }
                 }
             }
-        }
-        .environment(\.defaultMinListRowHeight, 0)
-        .frame(maxWidth: .infinity, maxHeight: .infinity)
-        .onAppear {
-            // When environment values are available through the view hierarchy:
-            //  - If we get a non-default theme through the environment values,
-            //    we use to override the theme from this view and its hierarchy.
-            //  - Otherwise we just refresh the tokens to reflect the theme
-            //    associated with the window that this View belongs to.
-            if theme == ThemeKey.defaultValue {
-                self.tokens.updateForCurrentTheme()
-            } else {
-                self.tokens.theme = theme
+            .environment(\.defaultMinListRowHeight, 0)
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .onAppear {
+                // When environment values are available through the view hierarchy:
+                //  - If we get a non-default theme through the environment values,
+                //    we use to override the theme from this view and its hierarchy.
+                //  - Otherwise we just refresh the tokens to reflect the theme
+                //    associated with the window that this View belongs to.
+                if theme == ThemeKey.defaultValue {
+                    self.tokens.updateForCurrentTheme()
+                } else {
+                    self.tokens.theme = theme
+                }
             }
-        }
     }
 }
 
 extension MSFListView {
     /// View for List Cells
     struct MSFListCellView: View {
-        var cell: MSFListCellData
+        var cell: MSFListCellState
```

Comment:
`@ObservedObject var state: MSFListCellState`

The collapse/expand seems to be broken as of the last iteration because this property is not using the @observedobject wrapper.